python/cudf/cudf/core/buffer.py

```diff
@@ -60,10 +60,19 @@ def __init__(self, data=None, size=None, owner=None):
     def __reduce__(self):
         return self.__class__, (self.to_host_array(),)
 
+    @property
+    def __cuda_array_interface__(self):
+        intf = {
+            "data": (self.ptr, False),
+            "shape": (self.size,),
+            "strides": (1,),
+            "typestr": "|u1",
```

Comment:
I typed this as `|u1` to align with `bytes` in Python and `rmm.DeviceBuffer`, which both use this type. That said, it appears `|i1` was being used before (i.e. signed instead of unsigned). Not sure if there was a reason for this. So please let me know if that typing is important.

Reply:
Looks like this is actually not insignificant. There are a few failures due to `gdf_dtype_from_dtype` not being able to handle unsigned values (see error snippet below). So have switched this to `|i1`, which should help. Can revisit in the future as needed 🙂

```python
    c_dtype = gdf_dtype_from_dtype(col.dtype)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   raise TypeError('cannot convert numpy dtype `%s` to gdf_dtype' % (dtype))
E   TypeError: cannot convert numpy dtype `uint8` to gdf_dtype
```